maze_transformer/utils/token_utils.py

```diff
@@ -0,0 +1,66 @@
+import typing
+
+from maze_transformer.generation.constants import SPECIAL_TOKENS
+
+
+def tokens_between(tokens: list[str], start_value: str, end_value: str) -> list[str]:
+    start_idx = tokens.index(start_value) + 1
+    end_idx = tokens.index(end_value)
+
+    return tokens[start_idx:end_idx]
+
+
+def get_adjlist_tokens(tokens: list[str]) -> list[str]:
+    return tokens_between(
+        tokens, SPECIAL_TOKENS["adjlist_start"], SPECIAL_TOKENS["adjlist_end"]
+    )
+
+
+def get_path_tokens(tokens: list[str]) -> list[str]:
+    """The path is considered everything from the first path coord to the end of the list, including the path_end token (ie everything we are asking the model to predict)"""
+    start_idx = tokens.index(SPECIAL_TOKENS["path_start"]) + 1
+    return tokens[start_idx:]
```

Comment:
Hmm I'm torn here. I do see your point, and it would be elegant for everything in here to use `tokens_between`. And if `start_value` was also optional, it would generalize to `get_tokens_up_to_path_start` too. 

But I'm hesitant for a couple of reasons
- I don't want to add too much complexity to `tokens_between`
- It's called tokens_**between** - that doesn't really fit with optional start or end values
- `get_path_tokens` and `get_tokens_up_to_path_start` are fairly simple right now - the benefit of switching them to use `tokens_between` doesn't seem that big.

Right now I'm leaning towards leaving this as is. What do you think?